File: Python/ProdukElektronik.py
class ProdukElektronik:
    # Static list untuk menyimpan semua produk
    daftar_produk = []

    def __init__(self, id_produk, nama_produk, kategori, stok, harga):
        # atribut produk
        self.id_produk = id_produk
        self.nama_produk = nama_produk
        self.kategori = kategori
        self.stok = stok
        self.harga = harga

    # getter untuk setiap atribut
    def get_id_produk(self):
        return self.id_produk

    def get_nama_produk(self):
        return self.nama_produk

    def get_kategori(self):
        return self.kategori

    def get_stok(self):
        return self.stok

    def get_harga(self):
        return self.harga

    # setter untuk setiap atribut
    def set_id_produk(self, id):
        self.id_produk = id

    def set_nama_produk(self, nama):
        self.nama_produk = nama

    def set_kategori(self, kategori):
        self.kategori = kategori

    def set_stok(self, stok):
        self.stok = stok

    def set_harga(self, harga):
        self.harga = harga

    # method untuk menampilkan detail produk
    def tampilkan(self):
        print(f"ID: {self.id_produk}")
        print(f"Nama Produk: {self.nama_produk}")
        print(f"Kategori: {self.kategori}")
        print(f"Stok: {self.stok}")
        print(f"Harga: Rp. {self.harga}")

    @staticmethod
    def tambah_data():
        # input id produk
        id_produk = input("Masukkan ID Produk: ")

        # periksa apakah ID sudah ada
        for p in ProdukElektronik.daftar_produk:
            if p.get_id_produk() == id_produk:
                print(f"Error: Produk dengan ID {id_produk} sudah ada. Penambahan dibatalkan.")
                return # keluar dari method

        # input data produk baru
        nama_produk = input("Masukkan Nama Produk: ")
        kategori = input("Masukkan Kategori: ")
        stok = int(input("Masukkan Stok: "))
        harga = int(input("Masukkan Harga: "))

        # simpan ke list static
        ProdukElektronik.daftar_produk.append(ProdukElektronik(id_produk, nama_produk, kategori, stok, harga))
        print("Produk berhasil ditambahkan.")

    @staticmethod
    def tampilkan_data():
        # jika list kosong
        if not ProdukElektronik.daftar_produk:
            print("Data produk kosong.")
            return
        # tampilkan semua produk
        for p in ProdukElektronik.daftar_produk:
            p.tampilkan()
            print()

    @staticmethod
    def cari_data():
        # input id yang dicari
        id_cari = input("Masukkan ID yang dicari: ")

        # cari berdasarkan id
        for p in ProdukElektronik.daftar_produk:
            if p.get_id_produk() == id_cari:
                p.tampilkan()
                return
        print("Produk tidak ditemukan.")

    @staticmethod
    def update_data():
        # input id yang ingin diupdate
        id_update = input("Masukkan ID yang ingin diupdate: ")

        for p in ProdukElektronik.daftar_produk:
            if p.get_id_produk() == id_update:
                # menu update
                while True:
                    print("\n=== Menu Update Produk ===")
                    print("1. Update Nama")
                    print("2. Update Kategori")
                    print("3. Update Stok")
                    print("4. Update Harga")
                    print("5. Update Semua Sekaligus")
                    print("0. Selesai")
                    pilihan = input("Pilih: ")

                    # update sesuai pilihan
                    if pilihan == "1":
                        nama_baru = input("Masukkan Nama baru: ")
                        p.set_nama_produk(nama_baru)
                        print("Nama berhasil diupdate.")
                    elif pilihan == "2":
                        kategori_baru = input("Masukkan Kategori baru: ")
                        p.set_kategori(kategori_baru)
                        print("Kategori berhasil diupdate.")
                    elif pilihan == "3":
                        stok_baru = int(input("Masukkan Stok baru: "))
                        p.set_stok(stok_baru)
                        print("Stok berhasil diupdate.")
                    elif pilihan == "4":
                        harga_baru = int(input("Masukkan Harga baru: "))
                        p.set_harga(harga_baru)
                        print("Harga berhasil diupdate.")
                    elif pilihan == "5":
                        nama_baru = input("Masukkan Nama baru: ")
                        kategori_baru = input("Masukkan Kategori baru: ")
                        stok_baru = int(input("Masukkan Stok baru: "))
                        harga_baru = int(input("Masukkan Harga baru: "))
                        p.set_nama_produk(nama_baru)
                        p.set_kategori(kategori_baru)
                        p.set_stok(stok_baru)
                        p.set_harga(harga_baru)
                        print("Semua data berhasil diupdate.")
                    elif pilihan == "0":
                        print("Selesai update.")
                        break
                    else:
                        print("Pilihan tidak valid.")
                return
        print("Produk tidak ditemukan.")

    @staticmethod
    def hapus_data():
        # input id yang ingin dihapus
        id_hapus = input("Masukkan ID yang ingin dihapus: ")

        # cari produk sesuai id lalu hapus
        for i, p in enumerate(ProdukElektronik.daftar_produk):
            if p.get_id_produk() == id_hapus:
                ProdukElektronik.daftar_produk.pop(i)
                print("Produk berhasil dihapus.")
                return
        print("Produk tidak ditemukan.")

File: Java/ProdukElektronik.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class ProdukElektronik {
    //menyiapkan atribut yang dipakai
    private String idProduk;
    private String namaProduk;
    private String kategori;
    private int stok;
    private int harga;

    //static list untuk menyimpan semua produk 
    private static List<ProdukElektronik> daftarProduk = new ArrayList<>();

    //konstruktor untuk inisialisasi data produk
    public ProdukElektronik(String idProduk, String namaProduk, String kategori, int stok, int harga) {
        this.idProduk = idProduk;
        this.namaProduk = namaProduk;
        this.kategori = kategori;
        this.stok = stok;
        this.harga = harga;
    }

    //getter untuk mengambil nilai atribut
    public String getIdProduk() {
        return idProduk;
    }
    public String getNamaProduk() {
        return namaProduk;
    }
    public String getKategori() {
        return kategori;
    }
    public int getStok() {
        return stok;
    }
    public double getHarga() {
        return harga;
    }

    //setter untuk mengubah nilai atribut
    public void setIdProduk(String id) {
        this.idProduk = id;
    }
    public void setNamaProduk(String nama) {
        this.namaProduk = nama;
    }
    public void setKategori(String kategori) {
        this.kategori = kategori;
    }
    public void setStok(int stok) {
        this.stok = stok;
    }
    public void setHarga(int harga) {
        this.harga = harga;
    }

    //method untuk menampilkan detail produk
    public void tampilkan() {
        System.out.println("ID: " + this.idProduk);
        System.out.println("Nama Produk: " + this.namaProduk);
        System.out.println("Kategori: " + this.kategori);
        System.out.println("Stok: " + this.stok);
        System.out.println("Harga: Rp. " + this.harga);
    }

    //method untuk menambahkan produk baru
    public static void tambahData(Scanner input) {
        System.out.print("Masukkan ID Produk: ");
        String id = input.next();

        //cek apakah ID produk sudah ada
        for (ProdukElektronik p : daftarProduk) {
            if (p.getIdProduk().equals(id)) {
                System.out.println("Error: Produk dengan ID " + id + " sudah ada. Penambahan dibatalkan.");
                return; // keluar dari method
            }
        }

        //input data produk baru
        input.nextLine(); 
        System.out.print("Masukkan Nama Produk: ");
        String nama = input.nextLine();
        System.out.print("Masukkan Kategori: ");
        String kategori = input.nextLine();
        System.out.print("Masukkan Stok: ");
        int stok = input.nextInt();
        System.out.print("Masukkan Harga: ");
        int harga = input.nextInt();

        //simpan ke list
        daftarProduk.add(new ProdukElektronik(id, nama, kategori, stok, harga));
        System.out.println("Produk berhasil ditambahkan.");
    }

    //method untuk menampilkan semua produk
    public static void tampilkanData() {
        if (daftarProduk.isEmpty()) {
            System.out.println("Data produk kosong.");
            return;
        }
        for (ProdukElektronik p : daftarProduk) {
            p.tampilkan(); //ditampilkan sebanyak jumlah produk
            System.out.println();
        }
    }

    //method untuk mencari produk berdasarkan ID
    public static void cariData(Scanner input) {
        System.out.print("Masukkan ID yang dicari: ");
        String id = input.next();

        //cek apakah ada produk dengan ID yang sesuai
        for (ProdukElektronik p : daftarProduk) {
            if (p.getIdProduk().equals(id)) {
                p.tampilkan(); //tampilkan produk jika ditemukan
                return;
            }
        }
        System.out.println("Produk tidak ditemukan.");
    }

    //method untuk mengupdate data produk
    public static void updateData(Scanner input) {
        System.out.print("Masukkan ID yang ingin diupdate: ");
        String id = input.next();
        input.nextLine(); 

        //cari produk sesuai ID
        for (ProdukElektronik p : daftarProduk) {
            if (p.getIdProduk().equals(id)) {
                int pilihan;
                do {
                    //menu pilihan update disesuaikan dengan pilihn atribut
                    System.out.println("\n=== Menu Update Produk ===");
                    System.out.println("1. Update Nama");
                    System.out.println("2. Update Kategori");
                    System.out.println("3. Update Stok");
                    System.out.println("4. Update Harga");
                    System.out.println("5. Update Semua Sekaligus");
                    System.out.println("0. Selesai");
                    System.out.print("Pilih: ");
                    pilihan = input.nextInt();
                    input.nextLine(); 

                    switch (pilihan) {
                        case 1:
                            System.out.print("Masukkan Nama baru: ");
                            String nama = input.nextLine();
                            p.setNamaProduk(nama);
                            System.out.println("Nama berhasil diupdate.");
                            break;
                        case 2:
                            System.out.print("Masukkan Kategori baru: ");
                            String kategori = input.nextLine();
                            p.setKategori(kategori);
                            System.out.println("Kategori berhasil diupdate.");
                            break;
                        case 3:
                            System.out.print("Masukkan Stok baru: ");
                            int stok = input.nextInt();
                            p.setStok(stok);
                            System.out.println("Stok berhasil diupdate.");
                            break;
                        case 4:
                            System.out.print("Masukkan Harga baru: ");
                            int harga = input.nextInt();
                            p.setHarga(harga);
                            System.out.println("Harga berhasil diupdate.");
                            break;
                        case 5:
                            //update semua atribut sekaligus
                            System.out.print("Masukkan Nama baru: ");
                            String namaBaru = input.nextLine();
                            System.out.print("Masukkan Kategori baru: ");
                            String kategoriBaru = input.nextLine();
                            System.out.print("Masukkan Stok baru: ");
                            int stokBaru = input.nextInt();
                            System.out.print("Masukkan Harga baru: ");
                            int hargaBaru = input.nextInt();

                            p.setNamaProduk(namaBaru);
                            p.setKategori(kategoriBaru);
                            p.setStok(stokBaru);
                            p.setHarga(hargaBaru);
                            System.out.println("Semua data berhasil diupdate.");
                            break;
                        case 0:
                            System.out.println("Selesai update.");
                            break;
                        default:
                            System.out.println("Pilihan tidak valid.");
                    }
                } while (pilihan != 0);
                return;
            }
        }
        System.out.println("Produk tidak ditemukan.");
    }

    //method untuk menghapus data produk
    public static void hapusData(Scanner input) {
        System.out.print("Masukkan ID yang ingin dihapus: ");
        String id = input.next();

        //cari produk lalu hapus jika ada
        for (int i = 0; i < daftarProduk.size(); i++) {
            if (daftarProduk.get(i).getIdProduk().equals(id)) {
                daftarProduk.remove(i); //menggunakan remove
                System.out.println("Produk berhasil dihapus.");
                return;
            }
        }
        System.out.println("Produk tidak ditemukan.");
    }
}
